tests/unit/test_main.py

```diff
@@ -45,6 +46,32 @@ def test_get_usage(self, stdout):
         self.assertEqual(exitcode, 0)
         self.assertIn('usage:', output)
 
+    def test_get_erronous_cfg_defaults(self):
+        temp_file = tempfile.NamedTemporaryFile()
```

Comment:
I suspect that the use of tempfile.NamedTemporaryFile() is the cquse of the error on windows, becquse the file is open in read and write mode. After writing, you may have to to a seek(0) to rewind the file.
I suggest that you use instead:
conf = tempfile.mkstemp()
self.addCleanup(os.unlink, conf) # ensure file will be delete
with open(conf, 'w') as f:
    f.write(...) # write the conf
# now, execute the test on conf